rudder-core/src/main/scala/com/normation/rudder/repository/xml/LicenseRepositoryXML.scala

```diff
@@ -45,49 +44,46 @@ import scala.xml._
 import com.normation.rudder.exceptions._
 import java.io.FileNotFoundException
 import com.normation.inventory.domain.NodeId
+import net.liftweb.common.Box
+import net.liftweb.common.Full
+import net.liftweb.common.Failure
+import net.liftweb.common.Empty
 
 class LicenseRepositoryXML(licenseFile : String) extends LicenseRepository {
 
   val logger = LoggerFactory.getLogger(classOf[LicenseRepositoryXML])
 
-  val licenseMap = mutable.Map[String, NovaLicense]()
+  var licenseMap = Map[NodeId, NovaLicense]()
 
-
-  def findLicense(nodeId: NodeId): Option[NovaLicense] = {
-    licenseMap.get(nodeId.value)
-  }
-
-  def getAllLicense(): Seq[NovaLicense] = {
-    loadLicenses()
-    licenseMap.map(x => x._2).toSeq
-  }
-
-    def addLicense(license: NovaLicense): Option[NovaLicense] = {
-      logger.debug("Adding a license {}", license)
-      licenseMap.put(license.uuid, license)
-      saveLicenseFile()
-      findLicense(NodeId(license.uuid))
+  override def getAllLicense(): Box[Map[NodeId, NovaLicense]] = {
+    for {
+      _ <- loadLicenses()
+    } yield {
+      licenseMap
     }
+  }
 
-
-  private[this] def loadLicenses(): Unit = {
-    licenseMap.clear
-
-    logger.debug("Loading document {}", licenseFile)
-    val doc = loadLicenseFile()
-
-    for (elt <- (doc \\"licenses" \ "license")) {
-      logger.debug("Loading License")
-      val license = NovaLicense.parseXml(elt)
-      licenseMap.put(license.uuid, license)
+  private[this] def loadLicenses(): Box[Unit] = {
```

Comment:
i indeed fail to see the benefit of the map, a def returning the result would do the same, especially since we are now loading the licences at each generation